tck/features/SetAcceptance.feature

```diff
@@ -0,0 +1,332 @@
+
+  Scenario: Concatenate elements onto a list property
+    Given any graph
+    When executing query:
+      """
+      CREATE (a {foo: [1, 2, 3]})
+      SET a.foo = a.foo + [4, 5]
+      RETURN a.foo
+      """
+    Then the result should be:
+      | a.foo           |
+      | [1, 2, 3, 4, 5] |
+    And the side effects should be:
+      | +nodes      | 1 |
+      | +properties | 2 |
+
+  Scenario: Concatenate elements in reverse onto a list property
+    Given any graph
+    When executing query:
+      """
+      CREATE (a {foo: [3, 4, 5]})
+      SET a.foo = [1, 2] + a.foo
+      RETURN a.foo
+      """
+    Then the result should be:
+      | a.foo           |
+      | [1, 2, 3, 4, 5] |
+    And the side effects should be:
+      | +nodes      | 1 |
+      | +properties | 2 |
+
+  Scenario: Overwrite values when using +=
+    Given any graph
+    And having executed:
+      """
+      CREATE (:X {foo: 'A', bar: 'B'})
+      """
+    When executing query:
+      """
+      MATCH (n:X {foo: 'A'})
+      SET n += {bar: 'C'}
+      RETURN n
+      """
+    Then the result should be:
+      | n                         |
+      | (:X {foo: 'A', bar: 'C'}) |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Retain old values when using +=
+    Given any graph
+    And having executed:
+      """
+      CREATE (:X {foo: 'A'})
+      """
+    When executing query:
+      """
+      MATCH (n:X {foo: 'A'})
+      SET n += {bar: 'B'}
+      RETURN n
+      """
+    Then the result should be:
+      | n                         |
+      | (:X {foo: 'A', bar: 'B'}) |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Explicit null values in a map remove old values
+    Given any graph
+    And having executed:
+      """
+      CREATE (:X {foo: 'A', bar: 'B'})
+      """
+    When executing query:
+      """
+      MATCH (n:X {foo: 'A'})
+      SET n += {foo: null}
+      RETURN n
+      """
+    Then the result should be:
+      | n               |
+      | (:X {bar: 'B'}) |
+    And the side effects should be:
+      | +properties | 1 |
+
+  Scenario: Non-existent values in a property map are removed with SET =
+    Given any graph
+    And having executed:
+      """
+      CREATE (:X {foo: 'A', bar: 'B'})
+      """
+    When executing query:
+      """
+      MATCH (n:X {foo: 'A'})
+      SET n = {foo: 'B', baz: 'C'}
+      RETURN n
+      """
+    Then the result should be:
+      | n                         |
+      | (:X {foo: 'B', baz: 'C'}) |
+    And the side effects should be:
+      | +properties | 3 |
+
+  Scenario: Fail at runtime when the expression is neither a node nor a relationship
+    Given any graph
+    And parameters are:
+      | node | 42 |
+    When executing query:
+      """
+      SET (CASE WHEN true THEN {node} END).name = 'neo4j'
+      RETURN count(*)
+      """
+    Then a ArgumentError should be raised at runtime: RequiresNodeOrRelationship
```

Comment:
The problem is that the property access operator (`.`) does not work on the boolean value, right? That sounds to me like `IllegalArgumentValue`. A map would also be a valid value, and having `RequiresXorYorZ` type error details quickly gets out of hand I believe.